crates/configuration/src/version4/mod.rs

```diff
@@ -405,549 +419,93 @@ fn base_type_representations() -> BTreeMap<String, database::TypeRepresentation>
-        | metadata::TypeRepresentation::Json => {
-            query_engine_metadata::metadata::TypeRepresentation::Json
-        }
-        metadata::TypeRepresentation::Enum(v) => {
-            query_engine_metadata::metadata::TypeRepresentation::Enum(v)
-        }
-    }
-}
-
-fn convert_comparison_operator(
-    comparison_operator: metadata::ComparisonOperator,
-) -> query_engine_metadata::metadata::ComparisonOperator {
-    query_engine_metadata::metadata::ComparisonOperator {
-        operator_name: comparison_operator.operator_name,
-        operator_kind: convert_operator_kind(&comparison_operator.operator_kind),
-        argument_type: convert_scalar_type_name(comparison_operator.argument_type),
-        is_infix: comparison_operator.is_infix,
-    }
-}
-
-fn convert_operator_kind(
-    operator_kind: &metadata::OperatorKind,
-) -> query_engine_metadata::metadata::OperatorKind {
-    match operator_kind {
-        metadata::OperatorKind::Equal => query_engine_metadata::metadata::OperatorKind::Equal,
-        metadata::OperatorKind::In => query_engine_metadata::metadata::OperatorKind::In,
-        metadata::OperatorKind::Custom => query_engine_metadata::metadata::OperatorKind::Custom,
-    }
-}
-
-fn convert_composite_types(
-    composite_types: metadata::CompositeTypes,
-) -> query_engine_metadata::metadata::CompositeTypes {
-    query_engine_metadata::metadata::CompositeTypes(
-        composite_types
-            .0
-            .into_iter()
-            .map(|(k, composite_type)| {
-                (
-                    query_engine_metadata::metadata::CompositeTypeName(k),
-                    convert_composite_type(composite_type),
-                )
-            })
-            .collect(),
-    )
-}
-
-fn convert_composite_type(
-    composite_type: metadata::CompositeType,
-) -> query_engine_metadata::metadata::CompositeType {
-    query_engine_metadata::metadata::CompositeType {
-        type_name: composite_type.type_name,
-        schema_name: Some(composite_type.schema_name),
-        fields: composite_type
-            .fields
-            .into_iter()
-            .map(|(k, field)| (k, convert_composite_type_field_info(field)))
-            .collect(),
-        description: composite_type.description,
-    }
-}
-
-fn convert_composite_type_field_info(
-    field: metadata::FieldInfo,
-) -> query_engine_metadata::metadata::FieldInfo {
-    query_engine_metadata::metadata::FieldInfo {
-        field_name: field.field_name,
-        r#type: convert_type(field.r#type),
-        description: field.description,
-    }
+    Ok(parsed_config)
 }
 
-pub fn convert_tables(tables: metadata::TablesInfo) -> query_engine_metadata::metadata::TablesInfo {
-    query_engine_metadata::metadata::TablesInfo(
-        tables
-            .0
-            .into_iter()
-            .map(|(k, table_info)| (k, convert_table_info(table_info)))
-            .collect(),
-    )
-}
+pub async fn write_parsed_configuration(
```

Comment:
So we have a parsed configuration, that is one after `ndc-postgres-cli update` and writes the parsed configuration into the new directory, as well as copying the native query files from the source dir to the out dir?